From 3e1a4ef7099dcb96b0918697584d83072ac233de Mon Sep 17 00:00:00 2001
From: Iurii Konovalenko <iurii.konovalenko@globallogic.com>
Date: Wed, 22 Jun 2016 17:28:18 +0300
Subject: [PATCH] xen: arm: Force to allocate Dom0 only in low memory

On board that has memory mapped over 4GB 64bit Xen may allocate
Dom0 memory from over 4GB space. But some of those boards
doesn't has 48bit DMA controller and some device drivers
might not work properly.
This patch force to allocate Dom0 only in low memory.

Signed-off-by: Iurii Konovalenko <iurii.konovalenko@globallogic.com>
---
 xen/arch/arm/domain_build.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xen/arch/arm/domain_build.c b/xen/arch/arm/domain_build.c
index 5c8aad3..c5e617b 100644
--- a/xen/arch/arm/domain_build.c
+++ b/xen/arch/arm/domain_build.c
@@ -354,7 +354,7 @@ static void allocate_memory_11(struct domain *d, struct kernel_info *kinfo)
 #endif
     int i;
 
-    bool_t lowmem = is_32bit_domain(d);
+    bool_t lowmem = 1;
     unsigned int bits;
 
     printk("Allocating 1:1 mappings totalling %ldMB for dom0:\n",
-- 
1.9.1

